examples/smartContracts/deployer.ts

```diff
@@ -118,15 +118,15 @@ export const deploySmartContracts = async (
           new Uint8Array(contract.args.serialize()),
         );
       }
-      if (contract.coins.rawValue().isGreaterThan(0)) {
+      if (contract.coins.getValue().isGreaterThan(0)) {
         datastore.set(
           new Uint8Array(
             new Args()
               .addU64(BigInt(i + 1))
               .addUint8Array(u8toByte(1))
               .serialize(),
           ),
-          u64ToBytes(BigInt(contract.coins.toNumber())), // scaled value to be provided here
+          u64ToBytes(BigInt(contract.coins.toCoins().getValue().toNumber())), // scaled value to be provided here
```

Comment:
so what happen if the amount doesnt fit on number ?. Here you are coverting a bignumber to number an then to bigInt.. You should use string instead of number.

Reply:
I fixed this part as you suggested @peterjah